Auth/Auth.Platform.iOS/AuthService.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using AeroGear.Mobile.Core;
+using AeroGear.Mobile.Core.Configuration;
+
+namespace AeroGear.Mobile.Auth
+{
+    public class AuthService : IAuthService
```

Comment:
If this is actually a platform-specific implementation, then we can just ask developers to init this. It can take a `MobileCore` instance as a parameter, and will register itself with the MobileCore. This way we don't need the `ServiceFinder` at all.

Reply:
@wei-lee the login page is in the `shared` project, in the login it does not have any reference to the platform specific AuthService module. It needs a way to get it. That's the reason why the ServiceFinder is there.

Reply:
@wei-lee nevermind: I got you now

Reply:
@ziccardi If that has become a problem, then we can think of introducing DI/IoC into our own lib. But it should not be forced upon the users of our SDK: they can still use whatever they want to use in their own applications.

Reply:
Why ask our developers to do something (init it themselves) if we can do it for them?

Because if we are asking them to init services we have to explain and educate how they all work together.  This way they can just say "give me auth for my platform" and it just works.  More importantly this way they can call "give me auth" from their shared code and the linking is performed/checked at compile time instead of the developer having to write an iOS "give me auth" and a Android "Give me auth".  Unless I am misunderstanding your comment.

Reply:
@secondsun The problem is that we can't do it automatically. Developers will have to tell us which implementation class to init any way. If you look at the implementation here, all we really did is to call the `Init` method of a service for developers, but developers have to tell us which implementation class to use. It also makes it awkward if developers want to use their own IoC to hold the instance.

If this is case, why not just ask them to do:

```
var auth = new AuthService(mobileCore);
```

and they will be able to easily add that instance to their own IoC. 

It's a standard constructor pattern and I don't really think a lot of education is required.